issue1-seed-identification-api-for-nachet-frontend.py

```diff
@@ -0,0 +1,126 @@
+
+    Args:
+        system_prompt (str): A system prompt for the OpenAI conversation.
+        user_prompt (str): A user prompt for the OpenAI conversation.
+        json_template (json): A JSON template for the OpenAI request.
+
+    This function performs the following steps:
+    1. Iterates through a list of seed values.
+    2. Checks if a JSON file for each seed exists and skips if it does.
+    3. Constructs an SQL query to retrieve data related to the seed from a database.
+    4. Sends the query to the database and fetches the retrieved data.
+    5. Concatenates the cleaned content into a single 'page.'
+    6. Sends a request to the Azure OpenAI endpoint to get a response.
+    7. Processes the response, extracting the name and saving it as a JSON file.
+    """
+    for url, seed_name in url_to_seed_mapping.items():
+        print("\nCurrent seed : " + seed_name)
+        seed_json_path = seed_name + ".json"
+
+        file_path = os.path.join(SEED_DATA_PATH, seed_json_path)
+    
+        if (os.path.exists(file_path)):
+            print(f"JSON file {seed_json_path} exists in {SEED_DATA_PATH}, skipping")
+        else:
+            web_pages = query_get_webpage(cursor, url)
+
+            all_language_seed_page = ""
+            for row in web_pages:
+                web_text = row.get("cleaned_content")
+                all_language_seed_page += web_text
+            page = all_language_seed_page
+            md5hash = row.get("md5hash")
+
+            ### Get the images corresponding to the current page
+            images_fetch = query_get_images(cursor, md5hash)
+
+            image_information = ""
+
+            for row in images_fetch:
+                image_links = row["photo_link"]
+                image_descriptions = row["photo_description"]
+                image_information += f"Image link: {image_links}"
+                image_information += f"\nImage description: {image_descriptions}\n\n"
+
+            print("Sending request for summary to Azure OpenAI endpoint...\n")
+
+            user_prompt = (
+                load_user_prompt
+                + "Return a JSON file that follows this template:\n\n"
+                + json_template
+                + "\n\nhere is the text to parse:\n"
+                + page
+                + "\n\nAnd here is the images descriptions:\n"
+                + image_information
+            )
+
+            response = openai.get_chat_answer(system_prompt, user_prompt, 2000)
+
+            data = json.loads(response.choices[0].message.content)
+
+            if isinstance(data, dict):
+                file_name = seed_name
+                file_name = nachet.decode_french_text(file_name)
+                file_name += ".json"
+
+            file_path = os.path.join(SEED_DATA_PATH, file_name)
+            with open(file_path, "w") as json_file:
+                json.dump(data, json_file, ensure_ascii=False, indent=4)
```

Comment:
Before writing the json to file, I would suggest a verification. For instance with a function that checks if all the requiered fields from the template are indeed present in each item of the dict. What do you think?

Reply:
It might be challenging because the fields can sometimes be empty, and there are instances where we genuinely lack the necessary information. I've requested the API to allow for these blank entries. If something is missing inadvertently, it usually signifies an error, and typically, we would catch that before recording it. So I don't thing hadding a verification is a good options.

Reply:
I see. However errors can still slip into the output. Like a typo in a field name. I guess it depends a lot on how tolerant the system that will use this function is to these errors. i suggest having a discussion about this with @rngadam. Marking this as resolved.